core/xml.js

```diff
@@ -332,6 +338,71 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
 };
 
 /**
+ * Decode an XML DOM and create blocks on the workspace.
+ * Append the new blocks at the bottom of the existing blocks
+ * @param {!Element} xml XML DOM.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.appendDomToWorkspace = function(xml, workspace) {
+  
+  // find bottom position and alignment of the current blocks  
+  //var blocks = workspace.getAllBlocks();
+  var blocks = workspace.getTopBlocks();
+  var offsetY = 0; // container for the y of the bottom block
+  var offsetX = 0; 
+  var blocksCount = blocks.length;
+  if (blocksCount > 0) {
+	// initialize the position of the corners using position 
+	// of the first block
+	var bHW = (("height" in blocks[0])? blocks[0].getHeightWidth() : {height: 0, width: 0});
```

Comment:
I think the issue is alignment. I see XXX case (assuming LTR):

 * The left-most existing block and the left-most new block lack an output connector. Then, all blocks align to that left edge.
 * The left-most existing block and the left-most new block both have an output connector (and no other top block is within TAB_WIDTH of that left edge). Then, all blocks also align to the same left edge.
 * The left-most existing block has a output connector, and new blocks do not.  Then, shift new blocks right by TAB_WIDTH.
 * The left-most existing block does not have an output connector, but the new blocks do. Then, shift the new blocks left by TAB_WIDTH.
 * The left-most existing block has a tab, but an also another existing block without a tab is within TAB_WIDTH. Then.... ?
 * The left-most new block has a tab, but the also another new block without a tab is within TAB_WIDTH. Then...?

I would accept a pull request that doesn't handle the last two case.